pkg/providers/vsphere/validator.go

```diff
@@ -90,7 +90,31 @@ func (v *Validator) ValidateVCenterConfig(ctx context.Context, datacenterConfig
 	return nil
 }
 
-// TODO: dry out machine configs validations.
+func (v *Validator) validateMachineConfigTagsExist(ctx context.Context, machineConfigs []*anywherev1.VSphereMachineConfig) error {
+	tags, err := v.govc.ListTags(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if tags exists in vSphere: %v", err)
+	}
+
+	tagIds := make([]string, 0, len(tags))
+	for _, t := range tags {
+		tagIds = append(tagIds, t.Id)
```

Comment:
I mean if this was it for tags, the strings would have worked, no?

Reply:
No because there are other areas looping on tag name whereas I need Id so I exposed whole tag struct so each case can get the value it needs.